src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -216,7 +216,9 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
                     string featureFlagIdBase64 = Convert.ToBase64String(featureFlagIdHash);
 
-                    StringBuilder featureFlagIdBuilder = new StringBuilder(featureFlagIdBase64[featureFlagIdBase64.Length - 1] == '=' ? featureFlagIdBase64.Length - 1 : featureFlagIdBase64.Length);
+                    int indexOfEquals = featureFlagIdBase64.IndexOf("=");
+
+                    StringBuilder featureFlagIdBuilder = new StringBuilder(indexOfEquals != -1 ? indexOfEquals : featureFlagIdBase64.Length);
 
                     for (int i = 0; i < featureFlagIdBuilder.Capacity; i++)
```

Comment:
Given string builder capacity varies based off characters appended to it (auto-expansion) I would stay away from using it in a for loop. Instead, prefer a clearly static value.